addons/controllers/antreaconfig_controller_test.go

```diff
@@ -207,6 +211,145 @@ var _ = Describe("AntreaConfig Reconciler and Webhooks", func() {
+
+			clusterKey := client.ObjectKey{
+				Namespace: clusterNamespace,
+				Name:      clusterName,
+			}
+			configKey := client.ObjectKey{
+				Namespace: clusterNamespace,
+				Name:      configCRName,
+			}
+
+			cluster := &clusterapiv1beta1.Cluster{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, clusterKey, cluster)
+				return err == nil
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			config := &cniv1alpha1.AntreaConfig{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, configKey, config)
+				if err != nil {
+					return false
+				}
+
+				// Check owner reference
+				if len(config.OwnerReferences) == 0 {
+					return false
+				}
+
+				Expect(len(config.OwnerReferences)).Should(Equal(1))
+				Expect(config.OwnerReferences[0].Name).Should(Equal(clusterName))
+
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.TrafficEncapMode).Should(Equal("encap"))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.AntreaTraceflow).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.AntreaPolicy).Should(Equal(true))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.FlowExporter).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.AntreaIPAM).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.ServiceExternalIP).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.Multicast).Should(Equal(false))
+				Expect(config.Spec.AntreaNsx.Enable).Should(Equal(true))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				cluster := &clusterapiv1beta1.Cluster{}
+				err := k8sClient.Get(ctx, clusterKey, cluster)
+				if err != nil {
+					return false
+				}
+
+				serviceCIDR, serviceCIDRv6, err := util.GetServiceCIDRs(cluster)
+				if err != nil {
+					return false
+				}
+
+				infraProvider, err := util.GetInfraProvider(cluster)
+				if err != nil {
+					return false
+				}
+
+				// Check infraProvider values
+				Expect(infraProvider).Should(Equal("vsphere"))
+
+				// Check ServiceCIDR and ServiceCIDRv6 values
+				Expect(serviceCIDR).Should(Equal("192.168.0.0/16"))
+				Expect(serviceCIDRv6).Should(Equal("fd00:100:96::/48"))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: clusterNamespace,
+					Name:      util.GenerateDataValueSecretName(clusterName, constants.AntreaAddonName),
+				}
+				secret := &v1.Secret{}
+				err := k8sClient.Get(ctx, secretKey, secret)
+				if err != nil {
+					return false
+				}
+				Expect(secret.Type).Should(Equal(v1.SecretTypeOpaque))
+
+				// check data value secret contents
+				secretData := string(secret.Data["values.yaml"])
+
+				Expect(strings.Contains(secretData, "serviceCIDR: 192.168.0.0/16")).Should(BeTrue())
```

Comment:
the reason to use string here is because the antrea config types in antreaconfig_util.go are private structs, we can not use them directly here, do you think it ok to make the structs public, or declare new structs in the test files? @knabben

Reply:
Golang yaml can Unmarshal yaml to unstructured map. We can use Unmarshal to get a map[interface{}]interface{} (see in https://github.com/go-yaml/yaml/blob/v3/README.md example code). This avoid copying the structs and making the structs public.

On the other hand, I think we can make the stucts public. The fields in the structs are already public.

Reply:
@edwardbadboy @knabben I tried unstructured map, it makes the case much more complicate, for example the values.yml is 
```
infraProvider: vsphere
antrea:
  config:
    serviceCIDR: 198.51.100.0/12
    trafficEncapMode: encap
    tlsCipherSuites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384
    disableUdpTunnelOffload: false
    featureGates:
      AntreaProxy: true
      EndpointSlice: true
      AntreaPolicy: true
      FlowExporter: false
      Egress: false
      NodePortLocal: false
      AntreaTraceflow: true
      NetworkPolicyStats: false
      AntreaIPAM: false
      ServiceExternalIP: false
      Multicast: false
      Multicluster: false
      SecondaryNetwork: false
      TrafficControl: false
```
if we want to use AntreaPolicy, we have to use three nested maps, which means:
yaml.Unmarshal(ret, &m)
antrea := m["antrea"].(map[interface{}]interface{})
config := antrea["config"].(map[interface{}]interface{})
features := config["featureGates"].(map[interface{}]interface{})

and even in this condition, features["AntreaPolicy"] is interface type, we should convert it to bool value to judge whether it is true or false.

what's more we also need to add antreaNsx fields here after antrea nsxt packages are included into antrea.